File: go/booking-up-for-beauty/booking_up_for_beauty.go
package booking

import (
	"fmt"
	"time"
)

// Schedule returns a time.Time from a string containing a date.
func Schedule(date string) time.Time {
	t, _ := time.Parse("1/2/2006 15:04:05", date)
	return t
}

// HasPassed returns whether a date has passed.
func HasPassed(date string) bool {
	t, _ := time.Parse("January 1, 2006 15:04:05", date)
	return time.Now().After(t)
}

// IsAfternoonAppointment returns whether a time is in the afternoon.
func IsAfternoonAppointment(date string) bool {
	t, _ := time.Parse("Monday, January 1, 2006 15:04:05", date)
	return t.Hour() >= 12 && t.Hour() < 18
}

// Description returns a formatted string of the appointment time.
func Description(date string) string {
	a := Schedule(date)
	return fmt.Sprintf("You have an appointment on %s, %s.",
		a.Weekday(),
		a.Format("January 2, 2006, at 15:04"))
}

// AnniversaryDate returns a Time with this year's anniversary.
func AnniversaryDate() time.Time {
	return Schedule("9/15/2023 00:00:00")
}

File: csharp/booking-up-for-beauty/BookingUpForBeauty.cs
using System;

static class Appointment
{
    public static DateTime Schedule(string appointmentDateDescription) =>
        DateTime.Parse(appointmentDateDescription);

    public static bool HasPassed(DateTime appointmentDate) =>
        DateTime.Now.Subtract(appointmentDate).TotalMilliseconds > 0;

    public static bool IsAfternoonAppointment(DateTime appointmentDate) => 
        appointmentDate.Hour >= 12 && appointmentDate.Hour < 18;

    public static string Description(DateTime appointmentDate) => 
        $"You have an appointment on {appointmentDate.ToString()}.";

    public static DateTime AnniversaryDate() =>
        new DateTime(DateTime.Now.Year, 9, 15, 0, 0 ,0);
}
